### ISSUE TITLE
Process Control Protocol Server doesn't handle missed checked in from hybernation

### ISSUE THREAD

**Issue by blakerouse** [elastic/elastic-agent#445]
Currently the logic in the GRPC server-side of the Elastic Agent that tracks the check-in of processes uses timestamps. This works fine until the computer is hybernated. Once hybernated the time skew will be large, so large that on wake the Elastic Agent will mark all processes as FAILED.

https://github.com/elastic/elastic-agent/blob/main/pkg/core/server/server.go#L890

This needs to be fixed to use a counter based system that way in the case of missed checked-ins its based on the number missed versus a timestamp.

For confirmed bugs, please report:
- Version: All
- Operating System: All
- Steps to Reproduce:

1. Run the Elastic Agent
2. Hybernate the computer for a few minutes
3. Wake the computer
4. View the Elastic Agent logs it should show that all the applications missed its last check-in and that it marks them failed

**Comment by jlind23** [comment]
@ycombinator I do not think this is a problem anymore, can we close?

**Comment by ycombinator** [comment]
@blakerouse Please keep me honest here but looking at the code for the command runtime around missed check-ins, it appears to be based on counters and timestamps both: https://github.com/elastic/elastic-agent/blob/e984ed2f75fd90733a6d5a318c0a43f52af9eb3f/pkg/component/runtime/command.go#L225-L230

Similar logic is used by the service runtime: https://github.com/elastic/elastic-agent/blob/e984ed2f75fd90733a6d5a318c0a43f52af9eb3f/pkg/component/runtime/service.go#L468-L473

So my understanding is that if a computer hibernates for more than `checkinPeriod`, we'll still count that as a missed check-in.

**Comment by blakerouse** [comment]
This should have been closed a longtime ago. This was fixed with v2 implementation for 8.6+. It is time-based but only the time-based is used to increment the counter. That means from hibernate its possible for the missed checkin to increment by 1, but then the check-in will happen within the next 30 second window and it gets reset.

This is all good and we have seen for more than a year now that we no longer get the issues we used to have pre-v2 implementation.
